portfolio/src/main/webapp/script.js

```diff
@@ -59,6 +59,35 @@ async function randomGreeting() {
     const msgContainer = document.getElementById("message-box");
     msgContainer.innerText = textResponse[choose];
 }
+function processTextInput() {
+    let val = document.getElementById("chat-input").value;
+    if(!val){
+        console.log('no input');
+    }else{
+        handleSendText(val);
+    }
+  }
+
+async function handleSendText(msg) {
+    console.log(msg);
+    let response = await fetch('/chat?' + new URLSearchParams({
+        text: msg,
+    }), {method : "POST"})
+    const textResponse = await response.json();
+    console.log(textResponse);
+    updateChatWall();
+    
+}
+
+async function updateChatWall(){
+    let response = await fetch('/chat')
+    const textResponse = await response.json();
+    deleteWall();
```

Comment:
instead of deleting then re-adding everything each time, it looks like it could be possible to use https://developer.mozilla.org/en-US/docs/Web/API/Document/childElementCount and only add the items that aren't previously added

Reply:
- solved, but if visitors want to see the chronically accurate chat order, they will have to use the refresh button.  Currently, just add whatever that visitor wrote down on that visitor screen without fetching real-time data from the server.